samples/sensor/qdec/sample.yaml

```diff
@@ -0,0 +1,15 @@
+sample:
+  description: Usage quadrature decoder sensor
+  name: qdec_sensor
+tests:
+  sample.sensor.qdec_sensor:
+    tags: sensors
+    platform_allow: nucleo_f401re
```

Comment:
samples under `samples/sensor` should not be board specific, please, filter based on a particular alias being enabled in this case. You can see the generic accelerometer sample as a reference.

Reply:
In case of generic accelerometer sample, we have in tree boards available with required hardware (accel sensors).
This won't be the case for current sensor and providing qdec configuration on random boards as part of their default configuration doesn't follow the board description guidelines (and first of all doesn't feel right anyway).
Current approach of using an overlay seems like a better approach. 

The unfortunate consequence is the restriction of the sample to boards for which an overlay is provided, as filtering based on a particular alias being enabled in this case is not possible in the case of configuration made using sample overlays (which is a "limitation" in the way twister applies filter).

I understand your request, but I don't see how to solve it easily.

Reply:
right, I always forget about this limitation in twister...